### PR TITLE
switch to rococo staging with HW mode

### LINKED ISSUE #144: AES key encryption causes enclave crash in HW mode
We found that encrypting using the user's shielding key (AES-GCM 256) in HW mode would cause the enclave to crash:
```
[2022-11-16T21:43:06Z ERROR integritee_service::sidechain_setup] Sgx(SGX_ERROR_ENCLAVE_CRASHED)
[2022-11-16T21:43:12Z ERROR integritee_service::sidechain_setup] Sgx(SGX_ERROR_ENCLAVE_CRASHED)
[+] Received finalized header update (33), syncing parent chain...
[+] Found 1 block(s) to sync
[2022-11-16T21:43:17Z ERROR integritee_service] Parentchain block syncing terminated with a failure: EnclaveApi(Sgx(SGX_ERROR_ENCLAVE_CRASHED))
```

The error is triggered by:
https://github.com/litentry/tee-worker/blob/f87dbb353490866f2c33bef19bb33c896c0574d7/app-libs/stf/src/helpers.rs#L132

It works in SW mode.

### PR DESCRIPTION
**Pull request by BillyWooo**
as title
Note: 
- [AES_GCM](https://docs.rs/aes-gcm/latest/aes_gcm/) doesn't work with HW mode. SW mode is fine.
- Replace [AES_GCM](https://docs.rs/aes-gcm/latest/aes_gcm/) to [RING](https://briansmith.org/rustdoc/ring/aead/index.html)
- adapt CI configs to use rococo parachain

resolved #144

### CODE REVIEW

File: app-libs/stf/src/helpers.rs [hunk 1]
```diff
@@ -119,21 +121,19 @@ pub fn set_block_number(block_number: u32) {
...
-	key: &UserShieldingKeyType,
-	data: &[u8],
-	aad: &[u8],
-	nonce: [u8; USER_SHIELDING_KEY_NONCE_LEN],
-) -> AesOutput {
-	let cipher = Aes256Gcm::new(key.into());
-	let payload = Payload { msg: data, aad };
-	let ciphertext = cipher.encrypt(&nonce.into(), payload).unwrap();
-	AesOutput { ciphertext, aad: aad.to_vec(), nonce }
+pub fn aes_encrypt(key: &UserShieldingKeyType, data: &[u8]) -> AesOutput {
```
Comment (Kailai-Wang) [inline app-libs/stf/src/helpers.rs]: I'd say we either:
- keep both `aes_encrypt_default` and `aes_encrypt` with the latter in a more generic form
- or consolidate them into one fn like  `aes_encrypt_default` where aad and nonce are not configurable.

File: local-setup/github-action-config-rococo.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
```
Comment (Kailai-Wang) [inline local-setup/github-action-config-rococo.json:1]: Can we keep `local-setup/github-action-config-one.json`? It's still useful for local testing

File: app-libs/stf/Cargo.toml [hunk 3]
```diff
@@ -9,6 +9,7 @@ edition = "2021"
 codec = { version = "3.0.0", default-features = false, features = ["derive"], package = "parity-scale-codec" }
 derive_more = { version = "0.99.5" }
 log = { version = "0.4", default-features = false }
+ring = { version = "0.16.20", default-features = false }
```
Comment (Kailai-Wang) [inline app-libs/stf/Cargo.toml]: Can you move it under `# litentry` and remove the `aes-gcm` dep?

**Review by Kailai-Wang — CHANGES_REQUESTED** [review]
Looks good, only a few small things.

And please submit a companion PR for parachain for the `ring` replacement

File: local-setup/github-action-config-rococo.json (hunk 2 above)
Comment (BillyWooo) [reply local-setup/github-action-config-rococo.json:1]: This is a redundant file. There is an identical one `github-action-config-one-worker.json`.

File: app-libs/stf/src/helpers.rs (hunk 1 above)
Comment (BillyWooo) [reply app-libs/stf/src/helpers.rs]: Done. Consolidate them together.

File: app-libs/stf/Cargo.toml (hunk 3 above)
Comment (BillyWooo) [reply app-libs/stf/Cargo.toml]: Done. 
`aes-gcm`, I tried to remove it before. Reported some strange dependency. Disappeared now.

**Review by Kailai-Wang — APPROVED** [review]

**Review by Kailai-Wang — COMMENTED** [review]
Just notice it's still litmus runtime?

**Comment by BillyWooo** [comment]
> Just notice it's still litmus runtime?

Yes. That's my last try with CI. Still working on rococo runtime.

**Pull request merged**